sdk/servicebus/azure_messaging_servicebus/src/client.rs

```diff
@@ -0,0 +1,559 @@
+// Copyright (c) Microsoft Corporation. All Rights reserved
+// Licensed under the MIT license.
+
+use crate::{
+    common::authorizer::Authorizer, ErrorKind, ReceiveMode, Receiver, Result, Sender,
+    ServiceBusError,
+};
+use azure_core::{credentials::TokenCredential, fmt::SafeDebug, http::ClientOptions, http::Url};
+use azure_core_amqp::{
+    AmqpConnection, AmqpConnectionApis, AmqpConnectionOptions, AmqpOrderedMap, AmqpSymbol,
+    AmqpValue,
+};
+use std::sync::Arc;
+
+/// SubQueue allows you to target a subqueue of a queue or subscription.
+///
+/// For example, the dead letter queue (SubQueueDeadLetter) or transfer dead letter queue (SubQueueTransfer).
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum SubQueue {
+    /// Targets the dead letter queue for a queue or subscription.
+    DeadLetter,
+    /// Targets the transfer dead letter queue for a queue or subscription.
+    Transfer,
+}
+
+impl SubQueue {
+    /// Returns the path suffix for the sub-queue.
+    pub fn as_path_suffix(&self) -> &'static str {
+        match self {
+            SubQueue::DeadLetter => "/$DeadLetterQueue",
+            SubQueue::Transfer => "/$Transfer/$DeadLetterQueue",
+        }
+    }
+}
+
+/// Options for configuring a Service Bus client.
+#[derive(Clone, SafeDebug)]
+pub struct ServiceBusClientOptions {
+    /// The API version to use when communicating with the Service Bus service.
+    pub api_version: String,
+
+    /// Core client configuration options.
+    pub client_options: ClientOptions,
```

Comment:
What fields from `client_options` do we use in ServiceBus? FWIW, EventHubs doesn't include it because there's nothing we use from it.